### DISCUSSION TITLE
Sdk refresh auth method send Autorization header in cookie mode

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#10071] [Archive]
Hi,

I'm trying to use as much as possible the sdk for authentication purpose from my frontend application. 

I have noticed the refresh method from the sdk is sending both: an Autorization Bearer token and the directus_refresh_token.
The Authorization header has the precedence over the cookie, therefore the refresh flow is not working properly in case the Autorization is expired.
The consequence is I cannot refresh the token with a valid directus_refresh_token cookie.

I saw it has been [explicitely handled](https://github.com/directus/directus/blob/278efcd3a364664c37b40a04190db34aed745ffe/app/src/auth.ts#L100) in the Directus app, the authorization header is removed for the refresh call. @rijkvanzanten Could you please clarify the intention ? How should it work when using the sdk refresh method?

@joselcvarela I saw you have worked on the Directus Transport layer refactoring. Do you have some hint about that. I wonder what would be the less destructive to remove this header before each request the sdk do. Also if possible I would like to not alterate the default transport.

Thanks for help

**Comment by joselcvarela** [comment]
Hello @gkielwasser 

>I have noticed the refresh method from the sdk is sending both: an Autorization Bearer token and the directus_refresh_token.
The Authorization header has the precedence over the cookie, therefore the refresh flow is not working properly in case the Autorization is expired.
The consequence is I cannot refresh the token with a valid directus_refresh_token cookie.

This seems to not be true on the latest version. Check this line https://github.com/directus/directus/blob/82547c004826e039caa38786dc9532cb31325ed6/packages/sdk/src/base/auth.ts#L88

Here we are clearing the previous token, then in here:
https://github.com/directus/directus/blob/82547c004826e039caa38786dc9532cb31325ed6/packages/sdk/src/base/directus.ts#L94-L98

Since there's no token the `Authorization` header is not set.

What version are you using?

**Reply by gkielwasser** [reply]
@joselcvarela  I see I'm using v9.0.0, and this line has been added after, I will update and check everything is fine. Thanks !

**Reply by gkielwasser** [reply]
I confirm the Authorization is not sent anymore. 
Perfect, thanks for those updates
